Partial and scheduled payments (#2)
hokodo-tech/documentation: docs/scheduled-payments.md

# Scheduled payments

## Introduction

If on your platform, buyers can pay sellers in multiple instalments it is very
likely that you will use the scheduled payments and the partial payments
features of the API.

Note that these features are completely optional and you can safely ignore if

> you don't intend to use it.

Reviewer: ```suggestion
you don't intend to use them.
```

This document is about defining a payemnt schedule. Declaring the schedule to
us will allow the seller to recover their payment earlier if the buyer can't
pay one instalment.

Partial payment feature is decribed in [this other section](partial-payments.md)
of the documentation.

## Concepts

A payment schedule is a list of ScheduledPayments or expected payments. Each

> represent an amount that the buyer is expected to pay at a given date. The sum

Reviewer: ```suggestion
represents an amount that the buyer is expected to pay at a given date. The sum
```

of the expected payments amounts will naturally be equal to the gross amount of
the transaction. ScheduledPayments are always understood in the same currency
as the transaction.

A schedule is provided with the creation of a transaction and cannot be modified.

## API reference

### Modified endpoint

#### `/v1/transactions`

A `scheduled_payments` field is added to `Transaction` objects. This field is a
list of `ScheduledPayment` objects.

### The ScheduledPayment object

attribute | type | flag | description
--------- | ---- | ---- | -----------
date | string(date) | mandatory | Expected due date of this instalment

> amount | decimal | mandatory | Expected amount of this instalment (positive in same currency as the transaction)

Reviewer: If this is a gross amount, we should say it here I think :o

### Validation

Field `scheduled_payments` is optional. If you provide it, fields

> `gross_amount`, `issue_date` and `due_date` of the transaction are mandatory.

Reviewer: ```suggestion
`gross_amount`, `issue_date` and `due_date` of the transaction become mandatory.
```

Coherence of the schedule is verified and first scheduled payment date shall

> be on or after `issue_date` and last scheduled payment date shall be on `due_date`.

Reviewer: ```suggestion
Coherence of the schedule is verified: first scheduled payment date shall
be on or after `issue_date`; last scheduled payment date shall be on `due_date`.
```
Or break into three sentences?
